cluster-autoscaler/apis/provisioningrequest/autoscaling.x-k8s.io/v1/types.go

```diff
@@ -113,6 +113,7 @@ type ProvisioningRequestSpec struct {
 	// +optional
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
 	// +kubebuilder:validation:MaxProperties=100
+	// +kubebuilder:validation:XValidation:rule="self.all(k, v, if parent.provisioningClassName == 'check-capacity.autoscaling.x-k8s.io' then k == 'noRetry' && v in ['true', 'false'] else true)",message="For check-capacity.autoscaling.x-k8s.io, only 'noRetry' parameter is supported with values 'true' or 'false'"
```

Comment:
I'm not sure we want to validate on the API level that only `noRetry` parameter is set.

My understanding is that the entire reason `Parameters` map exists is easy customisation for non-standard controllers, _without_ requiring an API changes. If we're going to introduce it on the API level, we might just as well add a new field for it (although in this case I would suggest using `maxRetries`, since it's more flexible). It'd also be cleaner than code in comment, especially with a potentially growing set of parameters/classes to allow.

@mwielgus can you PTAL and decide if this is the way to go or not? Since this is effectively an API level change, it would require SIG chair approval anyway. My opinion is that we shouldn't go with it.

Reply:
I would rather avoid adding explicit check for parameters. Parameters may be different for different classes and may come and go.